File: Internal/Context/Channel/GetMessagesContext.cs
using System.Text.Json.Serialization;

namespace BonfireServer.Internal.Context.Channel;

[Serializable]
public class GetMessagesContext : IBaseContext
{
    [JsonIgnore] public string? Token { get; set; }
    public byte Count { get; set; } = 64;
    public long ChannelId { get; set; }
    public bool Greedy { get; set; }
    public long Before { get; set; } = long.MaxValue;
}

File: Public/Scripts/Context/Channel/GetMessagesContext.ts
export class GetMessagesContext
{
    public Count: number;
    public ChannelId: number;
    public Greedy: boolean;
    public Before: number;

    constructor(
        channelId: number,
        count?: number,
        greedy?: boolean,
        before?: number,
    ) {
        this.Count = count ?? 64;
        this.ChannelId = channelId;
        this.Greedy = greedy ?? false;
        this.Before = before ?? 2**53;
    }
}